### ISSUE TITLE
Attachments not shown when addon is installed by policy

### ISSUE THREAD

**Issue by cm-schl** [TB-throwback/LookOut-fix-version#79]
* Operating System: Windows 10 with Thunderbird Policies to install addons (https://github.com/mozilla/policy-templates#extensionsettings)
* Thunderbird version number: 9.1.2
* 64 bit
* LookOut (fix version) version number: 4.2

# Current Behavior

When the addon is "force_installed" using the policies https://github.com/mozilla/policy-templates#extensionsettings no mail attachments a shown. In the error console this errors a visible:

```
Uncaught InternalError: too much recursion
    lo_get_pref chrome://lookout/content/lookout.js:78
    lo_get_bool_pref chrome://lookout/content/lookout.js:86
    lo_debug_check chrome://lookout/content/lookout.js:97
    on_end_all_attachments chrome://lookout/content/lookout.js:897
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
    on_end_all_attachments chrome://lookout/content/lookout.js:902
```

## Steps to Reproduce

In policies.json or in Windows Registry set the following:

```
{
"lookout@s3_fix_version": {
"installation_mode": "force_installed",
"install_url": "file:///c:/<path>/lookout@s3_fix_version.xpi",
"updates_disabled": true
}
}
```

**Comment by gumara-dev** [comment]
Same here. 

LookOut: Entering scan_for_tnef()
({message:({}), contentType:"application/vnd.oasis.opendocument.text", name:"xxx.odt", url:"imap://xxx@xxx.xxx.xxx:993/fetch%3EUID%3E/INBOX%3E54948?part=1.2&filename=xxx.odt", uri:"imap-message://xxx@xxx.xxx.org/INBOX#54948", isExternalAttachment:false, size:95183, sizeResolved:true, partID:"1.2"})
LookOut:    Content Type: 'application/vnd.oasis.opendocument.text'
LookOut:    Strict Content check failed
LookOut: debug enabled 
LookOut: Entering on_end_all_attachments()
LookOut: Entering scan_for_tnef()
({message:({}), contentType:"application/vnd.oasis.opendocument.text", name:"xxx.odt", url:"imap://xxx@xxx.xxx.xxx:993/fetch%3EUID%3E/INBOX%3E54948?part=1.2&filename=xxx.odt", uri:"imap-message://xxx@xxx.xxx.xxx/INBOX#54948", isExternalAttachment:false, size:95183, sizeResolved:true, partID:"1.2"})
LookOut:    Content Type: 'application/vnd.oasis.opendocument.text'
LookOut:    Strict Content check failed
LookOut: debug enabled 
LookOut: Entering on_end_all_attachments()
LookOut: Entering scan_for_tnef()

and so on....

**Comment by jobisoft** [comment]
> Because then the group policy says uninstall and then load Lookout again.

Are you sure this is what's happening? It would be a disaster, if the policy would always uninstall and re-install the extension, as that clears its local storage (independent of lookout-fixed).

**Comment by gumara-dev** [comment]
Sorry about the short message. 
Was in a meeting.
So even with a new profile, the same error occurs.
Seems to be a timing issue. Not every launch necessarily results in the error. Only when the message "Restoring HandleMultipleAttachments function to original" is missing from the log does the recursion occur.

TB version 91.2
Windows 10 64bit

**Comment by jobisoft** [comment]
Ok, so if the forced install does not match the actual ATN source URI, it re-installs every time. I have to look at that, but why DO you use the local file and not the true ATN url? If you switch, does it fix the error for you?

**Comment by dugite-code** [comment]
@jobisoft could this be a caused by the use of the wrapper APIs? If I remember correctly that is what triggers the load and unload of the extension. Although I don't see what reason using a file URI would be cause for re-installing the add-on each and every time.

**Comment by gumara-dev** [comment]
> 
> 
> Ok, so if the forced install does not match the actual ATN source URI, it re-installs every time. I have to look at that, but why DO you use the local file and not the true ATN url? If you switch, does it fix the error for you?

I think your suggestion with the ATN url should fix our problem.Haven't had time to test it yet.
We use the file links because we had a problem with group policies in the beginning. A workaround that is no longer needed.

**Comment by jobisoft** [comment]
[lookout_fix_version-beta-tb.zip](https://github.com/TB-throwback/LookOut-fix-version/files/7536919/lookout_fix_version-beta-tb.zip)

Can you check, if this version fixes this for you?

You have to restart twice. The first restart will still fail, because the old/installed add-on which is starting up and is not properly shutting down when the new version is installed over is causing the issue. On the second restart, the new version is starting up and should now be able to cope with the early kill.

**Comment by cm-schl** [comment]
Thanks a lot @jobisoft - works like a charm!
